SF iOS/SF iOS/Models/EventDataSource.m

```diff
@@ -104,6 +116,24 @@ - (Event *)eventAtIndex:(NSUInteger)index {
     return self.events[index];
 }
 
+/// Updates EventDataSoruce [Event] by text search or gets all events if there is no search term
+///
+/// - paramaters:
+///     -searchTerm: string to search
+/// - returns: void
+- (void)filterEventsWith:(NSString *)searchTerm {
+    if (searchTerm.length < 1) {
+        self.events = [Event allObjects];
+        [self.delegate reloadData];
+        return;
+    }
+    
+    NSPredicate *coffeeFilter = [NSPredicate predicateWithFormat:@"name CONTAINS[c] %@", searchTerm];
+    RLMResults *filteredCoffee = [Event objectsWithPredicate:coffeeFilter];
+    self.events = filteredCoffee;
```

Comment:
Setting events here and on line 126 above will break the notification AFAIK. That means any updates won't be reflected if the data source actually changes. I have to believe there is a way to just add a filter to the query. I'm looking into it.

Reply:
Very related: https://github.com/realm/realm-cocoa/issues/4674

Reply:
Ultimately the proper way to do this hasn't been implemented in realm so I think we have to change how we are doing it on our end. I propose we either:
a) add a shim layer as we discussed maybe in the slack
b) follow basically what Jeffrey spells out in his `Actual Results` section of the issue description.